contracts/upgradeable_contracts/erc20_to_native/InterestConnector.sol

```diff
@@ -0,0 +1,286 @@
+    }
+
+    /**
+     * @dev Tells configured address of the interest receiver.
+     * @param _token address of the invested token contract.
+     * @return address of the interest receiver.
+     */
+    function interestReceiver(address _token) public view returns (address) {
+        return addressStorage[keccak256(abi.encodePacked("interestReceiver", _token))];
+    }
+
+    /**
+     * Updates the interest receiver address.
+     * Only owner can call this method.
+     * @param _token address of the invested token contract.
+     * @param _receiver new receiver address.
+     */
+    function setInterestReceiver(address _token, address _receiver) external onlyOwner {
+        _setInterestReceiver(_token, _receiver);
+    }
+
+    /**
+     * @dev Pays collected interest for the specific underlying token.
+     * Requires interest for the given token to be enabled.
+     * @param _token address of the token contract.
+     */
+    function payInterest(address _token) external interestEnabled(_token) {
+        uint256 interest = _clampInterest(_token, interestAmount(_token));
+
+        _withdrawTokens(_token, interest);
+        _transferInterest(_token, interest);
+    }
+
+    /**
+     * @dev Tells the amount of underlying tokens that are currently invested.
+     * @param _token address of the token contract.
+     * @return amount of underlying tokens.
+     */
+    function investedAmount(address _token) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked("investedAmount", _token))];
+    }
+
+    /**
+     * @dev Internal function for transferring interest.
+     * Calls a callback on the receiver, if it is a contract.
+     * @param _token address of the underlying token contract.
+     * @param _amount amount of collected tokens that should be sent.
+     */
+    function _transferInterest(address _token, uint256 _amount) internal {
+        address receiver = interestReceiver(_token);
+        require(receiver != address(0));
+
+        ERC20(_token).transfer(receiver, _amount);
+
+        if (AddressUtils.isContract(receiver)) {
+            IInterestReceiver(receiver).onInterestReceived(_token);
+        }
+
+        emit PaidInterest(_token, receiver, _amount);
+    }
+
+    /**
+     * @dev Internal function for setting the amount of underlying tokens that are currently invested.
+     * @param _token address of the token contract.
+     * @param _amount new amount of invested tokens.
+     */
+    function _setInvestedAmount(address _token, uint256 _amount) internal {
+        uintStorage[keccak256(abi.encodePacked("investedAmount", _token))] = _amount;
+    }
+
+    /**
+     * @dev Invests all excess tokens.
+     * Requires interest for the given token to be enabled.
+     * @param _token address of the token contract considered.
+     */
+    function invest(address _token) public interestEnabled(_token) {
```

Comment:
May we have `invest()`(without parameters) method to be part of `ForeignBridgeErcToNative.sol`  which will simplify the dai tokens compounding and will reduce possible human errors for the investment action execution.

Reply:
Sure